railties/lib/rails/generators.rb

```diff
@@ -156,12 +156,25 @@ def self.invoke(namespace, args=ARGV, config={})
         args << "--help" if args.empty? && klass.arguments.any?(&:required?)
         klass.start(args, config)
       else
-        options     = sorted_groups.map(&:last).flatten
-        suggestions = options.sort_by {|suggested| levenshtein_distance(namespace.to_s, suggested) }.first(3)
-        msg =  "Could not find generator '#{namespace}'. "
-        msg << "Maybe you meant #{ suggestions.map {|s| "'#{s}'"}.join(" or ") }\n"
-        msg << "Run `rails generate --help` for more options."
-        puts msg
+        unless sorted_groups.group_by(&:size).min.first > namespace.length
+          options     = sorted_groups.map(&:last).flatten
+          suggestions = options.select do |option|
+            option.split(/[:_]/).any? { |option_segment|
+              namespace.to_s.split(/[:_]/).any? { |name_segment|
+                2 > levenshtein_distance(name_segment, option_segment)
```

Comment:
The original motivation was if your input was just 3 letter, it doesn't make sense to suggest something 2 char different (`dog` can never be a real world typo of `foo`). That's why it was 1, but I admit <kbd>migrate</kbd> (as opposed to `migration`) is a strong argument against it.

As I said, we can raise it depending on the length of words, or puts more weights on the header words (by that, <kbd>fix</kbd> is slightly closer to `foo` than <kbd>ceo</kbd> is, while both are 2 char different).

Still, we should figure out what are the most realistic and typical typos and then come up with a solid algorithm. Maybe it's something only likes of Google can do, and the question is do we really need to go so far here, and most Rails core staffs would say nay.